src/WikiService.Data/ArticleFileRepository.cs

```diff
@@ -12,5 +17,29 @@ public class ArticleFileRepository : IArticleFileRepository
     {
       _provider = provider;
     }
+    public async Task<bool> RemoveAsync(List<Guid> filesIds)
+    {
+      bool isRemoveSuccess = false;
+
+      if (filesIds == null)
+      {
+        return isRemoveSuccess;
+      }
+
+      if (await _provider
+        .ArticlesFiles
+        .AnyAsync(x => filesIds.Contains(x.FileId)))
+      {
+        _provider.ArticlesFiles.RemoveRange(
+          _provider.ArticlesFiles
+          .Where(x => filesIds.Contains(x.FileId)));
+
+        await _provider.SaveAsync();
+
+        isRemoveSuccess = true;
+      }
+
+      return isRemoveSuccess;
+    }
```

Comment:
may be:
    public async Task<bool> RemoveAsync(List<Guid> filesIds)
    {
      if (filesIds is null)
      {
        return false;
      }

      _provider.ArticlesFiles.RemoveRange(
        _provider.ArticlesFiles
        .Where(x => filesIds.Contains(x.FileId)));

      await _provider.SaveAsync();

      return true;
    }

Reply:
I rewrote the code in another way.